crates/corro-agent/src/agent/handlers.rs

```diff
@@ -447,11 +450,20 @@ pub async fn handle_changes(
             }
 
             debug!(count = %tmp_cost, "spawning processing multiple changes from beginning of loop");
-            join_set.spawn(util::process_multiple_changes(
-                agent.clone(),
-                bookie.clone(),
-                std::mem::take(&mut buf),
-            ));
+            let changes = std::mem::take(&mut buf);
+            let agent = agent.clone();
+            let bookie = bookie.clone();
+            let seen_mutex = seen_mutex.clone();
+            join_set.spawn(async move {
+                if let Err(e) = util::process_multiple_changes(agent, bookie, changes.clone()).await
+                {
+                    error!("could not process multiple changes: {e}");
+                    for change in changes {
+                        let v = *change.0.versions().start();
+                        seen_mutex.lock().remove(&(change.0.actor_id, v));
```

Comment:
Adding a mutex here is probably a bit harder than changing the return value of the future and handling it in the loop, in the `tokio::select!`. 

For example, you could return `BTreeMap<ActorId, RangeInclusiveSet<Version>>` (or something) from the future. It would represent the data to remove from the seen map. You'd only do this if the function errored, else you return an empty `BTreeMap` (does not allocate).